File: 算法导论/4/c/operation.c
#include <stdio.h>
#include <time.h>
#include <math.h>
#define n 10000000
#define v 121313
#define k 10

void f1(){
	int m = v;
	for(int i = 0; i < n; i++){
		m += k;
	}
}

void f2(){
	int m = v;
	for(int i = 0; i < n; i++){
		m *= k;
	}
}

void f3(){
	int m = v;
	for(int i = 0; i < n; i++){
		m /= k;
	}
}

void f4(){
	double m = v;
	for(int i = 0; i < n; i++){
		m = pow(v, k);
	}
}

void printTime(void (*f)()){
	clock_t t1,t2;
	t1 = clock();
	f();
	t2 = clock();
	float diff = (float)(t2 - t1) / CLOCKS_PER_SEC;
	printf("%f\n", diff);
}

int main(){
	printTime(f1);
	printTime(f2);
	printTime(f3);
	printTime(f4);
	return 0;
}

File: 算法导论/4/c/operation.js
var n = 10000000;
var v = 121313;
var k = 10;

console.log(n, v, k);

// 加法
function f1(){
	console.time("f1");
	var m = v;
	for(var i = 0; i < n; i++){
		m += k;
	}
	console.timeEnd("f1");
}

// 乘法
function f2(){
	console.time("f2");
	var m = v;
	for(var i = 0; i < n; i++){
		m *= v;
	}
	console.timeEnd("f2");
}

// 除法
function f3(){
	console.time("f3");
	var m = v;
	for(var i = 0; i < n; i++){
		m /= v;
	}
	console.timeEnd("f3");
}

// 乘方
function f4(){
	console.time("f4");
	var m = v;
	for(var i = 0; i < n; i++){
		Math.pow(m, k);
	}
	console.timeEnd("f4");
}

f1();
f2();
f3();
f4();
